src/hugr/region.rs

```diff
@@ -0,0 +1,289 @@
+        } = hugr;
+        Self {
+            root,
+            graph: FlatRegionGraph::new_flat_region(graph, hierarchy, root.index),
+            hierarchy,
+            op_types,
+        }
+    }
+}
+
+impl<'g> HugrView for FlatRegionView<'g> {
+    type Nodes<'a> = MapInto<portgraph::hierarchy::Children<'a>, Node>
+    where
+        Self: 'a;
+
+    type NodePorts<'a> = MapInto<<FlatRegionGraph<'g> as PortView>::NodePortOffsets<'a>, Port>
+    where
+        Self: 'a;
+
+    type Children<'a> = MapInto<portgraph::hierarchy::Children<'a>, Node>
+    where
+        Self: 'a;
+
+    type Neighbours<'a> = MapInto<<FlatRegionGraph<'g> as LinkView>::Neighbours<'a>, Node>
+    where
+        Self: 'a;
+
+    type PortLinks<'a> = MapWithCtx<
+        <FlatRegionGraph<'g> as LinkView>::PortLinks<'a>,
+        &'a Self,
+        (Node, Port),
+    > where
+        Self: 'a;
+
+    fn root(&self) -> Node {
+        self.root
+    }
+
+    fn get_parent(&self, node: Node) -> Option<Node> {
+        self.hierarchy
+            .parent(node.index)
+            .map(Into::into)
+            .filter(|&n| n == self.root)
+    }
+
+    fn get_optype(&self, node: Node) -> &OpType {
+        self.op_types.get(node.index)
+    }
+
+    fn node_count(&self) -> usize {
+        self.hierarchy.child_count(self.root.index) + 1
+    }
+
+    fn edge_count(&self) -> usize {
+        // Faster implementation than filtering all the nodes in the internal graph.
+        self.nodes()
+            .map(|n| self.output_neighbours(n).count())
+            .sum()
+    }
+
+    fn nodes(&self) -> Self::Nodes<'_> {
+        // Faster implementation than filtering all the nodes in the internal graph.
+        self.hierarchy.children(self.root.index).map_into()
+    }
+
+    fn node_ports(&self, node: Node, dir: Direction) -> Self::NodePorts<'_> {
+        self.graph.port_offsets(node.index, dir).map_into()
+    }
+
+    fn all_node_ports(&self, node: Node) -> Self::NodePorts<'_> {
+        self.graph.all_port_offsets(node.index).map_into()
+    }
+
+    fn linked_ports(&self, node: Node, port: Port) -> Self::PortLinks<'_> {
+        let port = self.graph.port_index(node.index, port.offset).unwrap();
+        self.graph
+            .port_links(port)
+            .with_context(self)
+            .map_with_context(|(_, link), region| {
+                let port = link.port();
+                let node = region.graph.port_node(port).unwrap();
+                let offset = region.graph.port_offset(port).unwrap();
+                (node.into(), offset.into())
+            })
+    }
+
+    fn num_ports(&self, node: Node, dir: Direction) -> usize {
+        self.graph.num_ports(node.index, dir)
+    }
+
+    fn children(&self, node: Node) -> Self::Children<'_> {
+        let mut iter = self.hierarchy.children(node.index).map_into();
+        if node != self.root {
+            // Eagerly empty the iterator.
+            // Ideally we would construct an empty iterator directly, but
```

Comment:
could you construct a new hierarchy with just one node and return the children of that?

Reply:
Unfortunately no because of the lifetime requirement :(

I may add a `Children::new_empty(&'a Hierarchy) -> Self<'a>` method, though it seems a bit hacky.

Reply:
if it's a hack define it as a non-member function locally?

Reply:
It requires accessing the internals, so it has to be done from petgraph's side.
https://github.com/CQCL/portgraph/pull/81